scala-native-crypto/src/java/security/SecureRandom.scala

```diff
@@ -0,0 +1,22 @@
+package java.security
+
+import scala.scalanative.unsafe._
+
+class SecureRandom() extends java.util.Random(0L) {
+
+  override def setSeed(x: Long): Unit = ()
+
+  override def nextBytes(bytes: Array[Byte]): Unit = Zone { implicit z =>
+    val len = bytes.length
+    val buffer = alloc[Byte](len)
```

Comment:
We can avoid `Zone` and copying memory by passing the pointer to the byte array directly via

Reply:
I'd rather not, unless that's an officially supported API (in which case, it should be exposed as a method :)

Reply:
It can't be removed without Scala Native breaking itself. I think it's safe to use. In case it will be removed in future Scala Native versions, we can adapt the code to use the newer alternatives.

Reply:
We can pattern match on subclasses perhaps? https://github.com/scala-native/scala-native/blob/main/nativelib/src/main/scala/scala/scalanative/runtime/Arrays.scala#L296

Reply:
To avoid the cast.

Reply:
I suppose a pattern match would be fine :)

Reply:
Well, I take that back. Basically my concern is if someone comes up with a new GC scheme, where GC managed memory is not "pinned" and can be moved around.

Reply:
At runtime it's the actual implementation of the array:

<img width="432" alt="image" src="https://user-images.githubusercontent.com/1052965/189343551-dce31b86-1e66-4082-a408-4a4b7682028c.png">

Reply:
If we feel so strongly about having something like this available, then I argue it should be exposed in Scala Native itself.

Reply:
> Well, I take that back. Basically my concern is if someone comes up with a new GC scheme, where GC managed memory is not "pinned" and can be moved around.

Wouldn't that break Scala Native first? https://github.com/scala-native/scala-native/blob/main/nativelib/src/main/scala/scala/scalanative/runtime/Arrays.scala#L167

IMO that's a very big IF.

Reply:
This pattern is used everywhere in the scala-native repository:
![immagine](https://user-images.githubusercontent.com/5793054/189343855-33ab84db-6a83-4cf8-b041-e0554cb7dc53.png)
Also, this library [does this already](https://github.com/lolgab/scala-native-crypto/blob/main/scala-native-crypto/src/java/security/MessageDigest.scala#L139).

Reply:
Lol, that's comforting 😝

I agree, it's probably fine in practice. But also this is a cryptography library. I'd rather not mess about.

Again, if this is a justified pattern, then let's make a public API and I'll happily make the change :)

Reply:
Yeah I think 

1. If you don't like the cast - pattern match
2. If you are worried about stability of public API changing (and it is public API) - that's IMO something that needs to be revisited when Scala Native breaks it, because that'd need a rewrite of a ton of stuff in scala native itself.

Reply:
> But also this is a cryptography library. I'd rather not mess about

The APIs on ByteArray are public.

Reply:
> The APIs on ByteArray are public.

Ah, I see your point. `Array[Byte]` can change, but `ByteArray` cannot. That seems good enough for me.

Reply:
When the match fails, you can use either your thing from CE with reusing a small stack buffer, or keep it simple alloc - I'm unaware of scenarious where a byte array will be anything other than `ByteArray` at runtime so it hopefully won't be visible.

That said, a rolling buffer would still be a good "what if" alternative to heap allocation 👍

Reply:
I don't think a library should work forever. If Scala Native breaks we adapt and release a new version. But it's not something I see happening.